### PR TITLE
Initial support and classes for Skaffold run configuration type

### PR DESCRIPTION
**Pull request by ivanporty**
Progress towards #12, #13.

Adds initial, empty implementation of Skaffold run configuration type extension, which will register future single/continuous run configuration factories and provide actual run configuration implementations.

For `runIde` target, pre-selects IDEA Community of the latest release version as a test IDE, making sure we're not targeting features of paid Ultimate version.

Initializes UI resources infrastructure (messages and icon) for Skaffold module. 
TODO: We need to request real Skaffold icon from their team. For now, generic Google Cloud Icon is used.

Result:
skaffold-initial-config

No tests for now as this is purely declaration code with almost no real functionality.

### CODE REVIEW

**Comment by ivanporty** [comment]
@etanshaul Please let me know how do we invite Kotlin team for our initial reviews?

**Comment by etanshaul** [comment]
@ivanporty there is a kotlin group that we can send messages to and request reviews. 

I suggest that we go through an internal review first to iron out any non language style things, and then bring in some Kotlin language reviewers.

I'll start looking at this.

**Comment by etanshaul** [comment]
g/kotlin-readability

File: build.gradle.kts [hunk 1]
```diff
@@ -11,4 +11,13 @@ allprojects {
 
     apply(plugin = "org.jetbrains.intellij")
     apply(plugin = "kotlin")
+
+    intellij {
+        type = "IC"
+        version = "2018.2"
```
Comment (etanshaul) [inline build.gradle.kts]: lets externalize this, in a separate PR. maybe an issue to track so we don't forget.

Comment (ivanporty) [reply build.gradle.kts]: #15

File: skaffold/src/main/kotlin/com/google/container/tools/skaffold/run/SkaffoldRunConfigurationType.kt [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+import com.intellij.execution.configurations.ConfigurationFactory
+import com.intellij.execution.configurations.ConfigurationType
+import javax.swing.Icon
+
+/**
+ * [ConfigurationType] extension that registers and enables Skaffold run configuration type, including single run and
+ * continuous deployment targets provided by separate [ConfigurationFactory].
+ */
+class SkaffoldRunConfigurationType : ConfigurationType {
+    val ID = "google-container-tools-skaffold-run-config"
```
Comment (etanshaul) [inline skaffold/src/main/kotlin/com/google/container/tools/skaffold/run/SkaffoldRunConfigurationType.kt]: question for Kotlin readibility reviewers: would we want this as a class parameter instead? i.e:

`class SkaffoldRunConfigurationType(val ID:String = "google-container-tools-skaffold-run-config")`

Or would this only be useful if someone needed to instantiate the class with a different ID (which would never happen in this case)?

File: skaffold/src/main/kotlin/com/google/container/tools/skaffold/SkaffoldUiResources.kt [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.container.tools.skaffold
+
+import com.intellij.CommonBundle
+import com.intellij.openapi.util.IconLoader
+import org.jetbrains.annotations.PropertyKey
+import java.util.*
```
Comment (tiembo) [inline skaffold/src/main/kotlin/com/google/container/tools/skaffold/SkaffoldUiResources.kt]: No wildcard imports

File: skaffold/src/main/kotlin/com/google/container/tools/skaffold/run/SkaffoldRunConfigurationType.kt [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+
+    override fun getIcon(): Icon {
```
Comment (tiembo) [inline skaffold/src/main/kotlin/com/google/container/tools/skaffold/run/SkaffoldRunConfigurationType.kt]: These overrides can be [expression functions](https://kotlinlang.org/docs/reference/functions.html#single-expression-functions)

File: skaffold/src/main/kotlin/com/google/container/tools/skaffold/run/SkaffoldRunConfigurationType.kt (hunk 2 above)
Comment (tiembo) [reply skaffold/src/main/kotlin/com/google/container/tools/skaffold/run/SkaffoldRunConfigurationType.kt]: That's correct - if the ID will never change, no need to have it as a class param.

File: skaffold/src/main/kotlin/com/google/container/tools/skaffold/SkaffoldUiResources.kt [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+
+internal val SKAFFOLD_ICON = IconLoader.getIcon("/icons/skaffold.png")
+
+private const val BUNDLE_NAME = "messages.SkaffoldBundle"
+
+/**
+ * Returns message by provided key from Skaffold message bundle with optional parameters.
+ */
+internal fun message(@PropertyKey(resourceBundle = BUNDLE_NAME) key: String, vararg params: String): String {
```
[the remark this answers is not in the document]
Comment (ivanporty) [reply skaffold/src/main/kotlin/com/google/container/tools/skaffold/SkaffoldUiResources.kt]: @tiembo what do you think about this file? It's kind of not obvious some class would make sense here so current implementation is just a couple of fields and function to retrieve UI messages, internal to the module. Does it fit better with Kotlin vs. Java where one always needs to come up with some utility class name?

File: skaffold/src/main/kotlin/com/google/container/tools/skaffold/run/SkaffoldRunConfigurationType.kt [hunk 6]
```diff
@@ -29,23 +29,13 @@ import javax.swing.Icon
...
 
-    override fun getIcon(): Icon {
-        return SKAFFOLD_ICON
-    }
+    override fun getIcon(): Icon = SKAFFOLD_ICON
 
-    override fun getConfigurationTypeDescription(): String {
-        return message("skaffold.run.config.general.description")
-    }
+    override fun getConfigurationTypeDescription(): String = message("skaffold.run.config.general.description")
```
Comment (etanshaul) [inline skaffold/src/main/kotlin/com/google/container/tools/skaffold/run/SkaffoldRunConfigurationType.kt]: is this over the line length limit (I think its 100 for java, I assume its the same for Kotlin)?

note to self: we should add the formatter and static analysis tools soon

Comment (ivanporty) [reply skaffold/src/main/kotlin/com/google/container/tools/skaffold/run/SkaffoldRunConfigurationType.kt]: Done, shortened.

File: skaffold/src/main/kotlin/com/google/container/tools/skaffold/run/SkaffoldRunConfigurationType.kt [hunk 7]
```diff
@@ -0,0 +1,43 @@
...
+
+/**
+ * [ConfigurationType] extension that registers and enables Skaffold run configuration type,
+ * including single run and continuous deployment targets provided by separate
+ * [ConfigurationFactory].
+ */
+class SkaffoldRunConfigurationType : ConfigurationType {
+    val ID = "google-container-tools-skaffold-run-config"
+
+    override fun getIcon(): Icon = SKAFFOLD_ICON
```
Comment (tiembo) [inline skaffold/src/main/kotlin/com/google/container/tools/skaffold/run/SkaffoldRunConfigurationType.kt]: It's also OK to omit the return type for expression functions, though helpful to leave them in if the return isn't obvious.  This isn't a strict rule, just whatever you think works best.

File: skaffold/src/main/kotlin/com/google/container/tools/skaffold/SkaffoldUiResources.kt (hunk 5 above)
[the remark this answers is not in the document]
Comment (tiembo) [reply skaffold/src/main/kotlin/com/google/container/tools/skaffold/SkaffoldUiResources.kt]: Totally fine to have [top level functions](https://kotlinlang.org/docs/reference/functions.html#function-scope) in Kotlin.

File: skaffold/src/main/kotlin/com/google/container/tools/skaffold/run/SkaffoldRunConfigurationType.kt (hunk 6 above)
Comment (tiembo) [reply skaffold/src/main/kotlin/com/google/container/tools/skaffold/run/SkaffoldRunConfigurationType.kt]: [ktlint](https://github.com/shyiko/ktlint) is a great linter for Kotlin (it's used in several of our Android DevRel projects).

**Pull request merged**